### ISSUE TITLE
just some feedback from a happy user :)

### ISSUE THREAD

**Issue by rado84-github** [CanisLupus/swift-selection-search#249]
I gotta say SSS never ceases to amaze me! :) I keep finding new ways to make my search life easier, all thanks to this extension. It's safe to say I can't live without SSS!
After Canis Lupus explained it to me how do a group search, now I found a new (lazy) way for quick search. For instance, I like playing most of the games with cheats, especially games that involve shooting, so I constantly have to search for cheat engine tables which means that every time I had to type the game title, "cheat tables" and the website where to search for that. If you do that once, it's OK, but if you have to do it for every game, it gets annoying quickly. So I decided to try and add stuff to the address AFTER {searchTerms} and it worked! :D 
https://duckduckgo.com/?q={searchTerms}+cheat+tables+site%3Aforum.cheatengine.org&t=ffab&atb=v318-7&ia=web
With this address I simply have to highlight the game title, right click on it and choose the relevant search name.
SSS is also useful (and frankly far more effective) for translating foreign text. :) Mostly because the translating extensions usually display things I don't wanna see or need to see, or they attemt translation in the same tab where the foreign text is, which, needless to say, makes a huge mess on the screen. But with SSS it's very simple. :D 
https://translate.google.com/?sl=auto&tl=bg&text={searchTerms}&op=translate

Basically, if the website has an input form which is reflected in the URL address, SSS can save you a lot of future typing. :D :D

**Comment by CanisLupus** [comment]
Hi! Thank you so much for the kind words and tips :D

I have tried showing some "tips" to users in the form of (optional) SSS's default engines (for example, there's one using translation to English) and under "advanced usage" in the instructions, but I didn't want to bloat the options page even more. I suspect that having an actual webpage of SSS search engines and usage tips would be a Good Thing™.

I'm not sure what to do about this "issue" hahah :) There is nothing to solve so I suppose it should be closed, shouldn't it? 😄 But I always appreciate reading stuff like this. Contributing to someone's time-savings is a great feeling!

Cheers!

**Comment by Aierbote** [comment]
I hope you don't mind if I list my suggestions here, because it seems on topic and I don't want to open an issue just for a couple of feedbacks.

I like the suggestions above and I will add them too.

I personally added two ""search engines"":
- one to point to the SSS settings (cause going to `about:addons`> Swift Selection Search > `...` > Preferences was boring every time) 

| Name | Search URL | Icon URL |
| :-- | :-- | :-- |
| Extension Settings | moz-extension://90e55aac-519a-534a-9372-a792ecf55cd9/settings/settings.html | URL to an cog wheel icon |

- one to just to hover it to read the shortcuts ""tooltip"" I simply put in the `Name` cell

| Name | Search URL | Icon URL |
| :-- | :-- | :-- |
| **toolTips** | about:blank | URL to an Info Point icon |

Where **toolTips** is a long string with lots of whitespaces in the middle (48, if you could check this comment in raw)
"""Open selection in popup =  `shift + cmd + space`                                                  Toggle auto popup opening = `shift + cmd + U`"""

But the latter is not a polish result, hard to simulate a new line (`\n`, `\t` and `<br>` don't work). Plus I wrote it **manually** but I think it could be better if somehow managed on the backend, so if an user changed the default shortcuts to something else by hovering on the tooltip it could read the variables where shortcuts are stored. A default tooltip button!

Furthermore if clicked on this official tooltip button could open directly your browser `Manage Shortcuts` (for example Firefox warns you if you are trying to set an already in use shortcut), instead of my `about:blank` workaround (which opens a new tab if I type it by mistake)

---

> I suspect that having an actual webpage of SSS search engines and usage tips would be a Good Thing™.

I completely agree, IMHO even just a well spaced README.md would be enough for the job

**Comment by CanisLupus** [comment]
Hey @Aierbote, sorry for the late reply! I think it's okay to post the suggestions here :) Although it's important to note that development of new features has stopped indefinitely.

It's great that you could come up with these with current SSS! The icon for the settings would have probably been a big time saver for me when developing hahah.

Adding these as defaults in SSS would unfortunately be a bit anti-pattern for users, since neither of them actually use the selected text (unlike all others in SSS). But on topic with my previous reply: it would be cool to have a page for all these user ideas. A page that I don't think I'd have time (or maybe just energy, actually) to maintain, sadly 😅
